src/srf/boolean.cpp

```diff
@@ -20,20 +20,25 @@ void SShell::MakeFromIntersectionOf(SShell *a, SShell *b) {
     MakeFromBoolean(a, b, SSurface::CombineAs::INTERSECTION);
 }
 
-// We will be inserting existing verticies into curves to split them
-// todo: this is only using the ends of exact curves, and it is only
-// using them to split existing curves, not new intersections.
-// It resolves some issues but we could do better. We will need to
-// reorder things so the surface intersection curves exist prior to
-// splitting any curves at all in order to have their verticies too.
-// Better still would be to get curve/surface intersection to work
-// more reliably at the edges - maybe do curve/curve tests as part
-// of the curve-surface intersection test.
-static void FindVertsOnCurve(List<SInter> *l, const SCurve *curve, SShell *sh) {
-    for(auto sc : sh->curve) {
-        if(!sc.isExact) continue;
-        for(int i=0; i<2; i++) {
-            Vector pt = sc.exact.ctrl[ i==0 ? 0 : sc.exact.deg ];
+// We will be inserting existing surface verticies into curves to split them.
+// This is helpful when curved surfaces become tangent along a trim and the
+// usual tests for curve-surface intersection don't split spllit the curve.
+// Another option might be to split at the trim curve verticies which can
+// handle more a few more cases. It's not clear if either is the best approach.
+static void FindVertsOnCurve(List<SInter> *l, const SCurve *curve, SShell *sh,
+                            Vector amin, Vector amax) {
+
+    for(auto srf : sh->surface) {
+        Vector bmax, bmin;
+        srf.GetAxisAlignedBounding(&bmax, &bmin);
+
+        if(Vector::BoundingBoxesDisjoint(amax, amin, bmax, bmin)) {
+            // They cannot possibly intersect, no curves to generate
+            continue;
+        }
+        
+        for(int i=0; i<4; i++) {
+            Vector pt = srf.ctrl[ (i&1)==0 ? 0 : srf.degm ][ (i&2)==0 ? 0 : srf.degn];
```

Comment:
So you are only checking whether the four corners of the surface lie on the curve. Not all control points.
This sounds logical for what you are trying to do, but the function should be called "FindSrfCornersOnCurve" or something.
And the comment above should be explicit about it.